src/main/java/var3/LeagueService.java

```diff
@@ -0,0 +1,23 @@
+package var3;
+
+import var3.abstracts.Team;
+
+public class LeagueService<E extends Team> {
+    private League<E> league;
```

Comment:
here you should have a map(that will represent your table) with key -> sport type and value -> the set with your teams